### ISSUE TITLE
Add examples with nested documents and arrays to `update`

### ISSUE THREAD

**Issue by mglukhovsky** [rethinkdb/docs#727]
From a user [on Twitter](https://twitter.com/jrychter/status/591213544244383744): 

> @rethinkdb your docs are great, lots of info on querying, but rather thin on update(). Need examples with complex strs (nested, arrays, etc)

Looking at the [documentation for update](http://rethinkdb.com/api/javascript/update/) I agree with that, we could use some examples of nested document updates and array updates.

**Comment by jwr** [comment]
To clarify (I'm the Twitter user @jrychter who tweeted this): I've spent lots of time with your documentation, which goes extensively into queries, but spends relatively little time on updates. Updates are important, because this is where you see lots of differences between a document model and a relational table-based model: especially if one uses a lot of nesting, updates become complicated.

I also posted a Stack Overflow question which is related to updating more complex documents: http://stackoverflow.com/questions/29821448/rethinkdb-how-to-append-to-an-array-in-a-nested-structure

I'd suggest thinking of complex examples: appends to arrays in nested fields, appends to nested fields found by filter(), updates that use subqueries, joins that expand nested documents, etc. — but to start with, even your "Ten-minute guide" could be updated (ahem) to include an example that shows how to update the "content" of a blog post by "William Adama" with the title of "We are at war".
